### PR TITLE
Process killer feature

### PR DESCRIPTION
**Pull request by udarrr**
@christian-bromann Generally it's a good feature when we can check and kill selenium processes by port :4444 and drivers names before starting.

Tests added

Bug is here https://github.com/webdriverio/selenium-standalone/issues/599

And I'm not convinced but maybe it have sense to be switched on in config by default. Currently it's not switched on.

### CODE REVIEW

**Comment by udarrr** [comment]
@christian-bromann Generally it's a good feature when we can check and kill selenium process by port :4444 with hanged drivers processes before starting.

File: docs/API.md [hunk 1]
```diff
@@ -111,6 +111,8 @@ By default all drivers are loaded, you only control and change the versions or a
 
 `opts.requestOpts` can be any valid [`got` options object](https://www.npmjs.com/package/got#proxies). You can use this for example to set a timeout.
 
+`opts.processKiller` set port of selenium server for killing, could be ;ile `4444` or `:4444`.
```
Comment (christian-bromann) [inline docs/API.md:114]: ;ile?

**Review by christian-bromann — COMMENTED** [review]
I am not sure how useful this would be for a user given they might not know on which port the driver was run. Can't we just improve how we kill the process and try another attempt if we we detect that the port is still occupied?

I think it would be a better approach to actually guarantee to free up the port.

**Comment by udarrr** [comment]
> I am not sure how useful this would be for a user given they might not know on which port the driver was run. Can't we just improve how we kill the process and try another attempt if we we detect that the port is still occupied?
> 
> I think it would be a better approach to actually guarantee to free up the port.

@christian-bromann Yes agreed it has sense, not necessary kill it if port is free already, i've added changes

File: docs/API.md [hunk 2]
```diff
@@ -111,6 +111,8 @@ By default all drivers are loaded, you only control and change the versions or a
 
 `opts.requestOpts` can be any valid [`got` options object](https://www.npmjs.com/package/got#proxies). You can use this for example to set a timeout.
 
+`opts.processKiller` set for that truthy value, for killing selenium server port.
```
Comment (christian-bromann) [inline docs/API.md:114]:
```suggestion
`opts.processKiller` set to a truthy value, for killing selenium server port.
```

File: docs/API.md [hunk 3]
```diff
@@ -121,6 +123,8 @@ If you're getting this error, it means that you didn't shut down the server succ
 pkill -f selenium-standalone
 ```
 
+or use truthy `opts.processKiller` in config
```
Comment (christian-bromann) [inline docs/API.md:126]:
```suggestion
or use `opts.processKiller` in config
```

File: docs/CLI.md [hunk 4]
```diff
@@ -49,6 +49,9 @@ selenium-standalone start --singleDriverStart=chrome
 selenium-standalone install --config=/path/to/config.json
 selenium-standalone start --config=./config/seleniumConfig.js
 
+# killing process before starting
+selenium-standalone start --processKiller=true
```
Comment (christian-bromann) [inline docs/CLI.md:53]: A user would ask why they would need to set this given it feels like the expected behavior is that the selenium server or driver shuts down completely when the process is triggered to get shutdown.

**Review by christian-bromann — COMMENTED** [review]
It still is a bit confusing to me why I would need to set `opts.processKiller` because as mentioned in the comment as a user I would expect that this package is able to shut down the selenium server process and clear the port automatically.

**Review by christian-bromann — APPROVED** [review]
Ok, sgtm 👍

**Pull request merged**